modules/sagemaker/sagemaker-templates-service-catalog/common/code_repo_construct.py

```diff
@@ -0,0 +1,293 @@
+            # Initialize Git repo and push
+            init_and_push_repo(repo_url, tmp_dir)
+
+            print(f"Repository initialized and pushed successfully")
+
+            # Retrieve the list of existing source credentials from codebuild
+            codebuild = boto3.client('codebuild')
+            response = codebuild.list_source_credentials()
+
+            existing_source_credentials = response['sourceCredentialsInfos']
+            # Check if a source credential for GitHub already exists
+            github_source_credential = any(cred['serverType'] == 'GITHUB' for cred in existing_source_credentials)
+            print(f"GitHub source credential already exists: {github_source_credential}")
+
+            if github_source_credential is False:
+                codebuild.import_source_credentials(
+                    token=code_connection_arn,
+                    serverType="GITHUB",
+                    authType="CODECONNECTIONS"
+                )
+                print(f"GitHub source credential imported successfully")
+            else:
+                print(f"GitHub source credential already exists")
+
+            response_data = {
+                'RepoName': repo_name,
+                'RepoUrl': f"https://github.com/{github_owner}/{repo_name}.git"
+            }
+
+            # Send a successful response back to CloudFormation
+            cfnresponse.send(event, context, cfnresponse.SUCCESS, response_data)
+
+
+    elif event['RequestType'] == 'Delete':
+        # Delete GitHub repository using GitHub API
+        headers = {
+            'Authorization': f'token {github_token}',
+            'Accept': 'application/vnd.github.v3+json',
+            'Content-Type': 'application/json'
+        }
+
+        req = urllib.request.Request(f"https://api.github.com/repos/{github_owner}/{repo_name}",
+                    headers=headers,
+                    method='DELETE')
+        try:
+            with urllib.request.urlopen(req) as response:
+                response_code = response.getcode()
+                if response_code == 204:
+                    print(f"Repository {repo_name} deleted successfully")
+        except urllib.error.HTTPError as e:
+            print(f"HTTP Error: {e.code} - {e.reason}")
+            print(e.read().decode())
+            raise
+
+        cfnresponse.send(event, context, cfnresponse.SUCCESS, {})
+    else:
+        # raise Exception(f"Invalid request type: {event['RequestType']}")
+        cfnresponse.send(event, context, cfnresponse.FAILED, {})
+"""
+        # Lambda function to create GitHub repository
+        github_repo_creator_lambda = lambdafunction.Function(
+            self,
+            "GitHubRepoCreatorLambda",
+            runtime=lambdafunction.Runtime.PYTHON_3_12,
+            handler="index.lambda_handler",
+            code=lambdafunction.Code.from_inline(github_lambda_func_code),
```

Comment:
Can you please move the handler code to a separate file ie `lambda/handler.py` and use `Code.from_asset`?

Reply:
I will take this up in next PR where other templates need to be integrated to GitHub repo